### PR TITLE
Product

### PR DESCRIPTION
**Pull request by isidzukuri**

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @item = Product.find(params[:id])
+    if @item.update(permited_params)
+      redirect_to @item
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @item = Product.find(params[:id])
```
Comment (BLauris) [inline app/controllers/products_controller.rb]: what will happen if smth. will went wrong and Product with [:id] will not be there?
there is nothing to delete!

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @item = Product.new
+  end
+
+  def create
+    authorize! :create, Product
+    @item = Product.new(permited_params)
+    if @item.save
+      redirect_to @item
+    else
+      render 'new'
```
Comment (BLauris) [inline app/controllers/products_controller.rb]: render :new

File: app/views/categories/show.html.haml [hunk 3]
```diff
@@ -2,5 +2,8 @@
   .panel-heading= @item.title
   .panel-body
     - if admin?
-      = link_to "Edit", edit_category_path, :class => 'btn btn-primary'
-      = link_to "Destroy", category_path(@item), method: :delete, :class => 'btn btn-danger'
\ No newline at end of file
+      = link_to "Edit", edit_category_path, class: 'btn btn-primary'
+      = link_to "Destroy", category_path(@item), method: :delete, class: 'btn btn-danger'
```
Comment (BLauris) [inline app/views/categories/show.html.haml]: add a alert here, that would ask "Are you sure?" so user wouldn't accidentally delete smth.

File: app/views/products/show.html.haml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+.panel.panel-default
+  .panel-heading= @item.title
+  .panel-body
+    - if admin?
+      = link_to "Edit", edit_product_path, class: 'btn btn-primary'
+      = link_to "Destroy", product_path(@item), method: :delete, class: 'btn btn-danger'
```
Comment (BLauris) [inline app/views/products/show.html.haml]: add a confirmation as I wrote above

Comment (isidzukuri) [inline app/views/products/show.html.haml]: added

File: app/views/categories/show.html.haml (hunk 3 above)
Comment (isidzukuri) [inline app/views/categories/show.html.haml]: added

File: app/controllers/products_controller.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    end
+  end
+
+  def update
+    @item = Product.find(params[:id])
+    if @item.update(permited_params)
+      redirect_to @item
+    else
+      render 'edit'
```
Comment (isidzukuri) [inline app/controllers/products_controller.rb]: done

File: app/controllers/products_controller.rb (hunk 2 above)
Comment (isidzukuri) [inline app/controllers/products_controller.rb]: done

File: app/controllers/products_controller.rb (hunk 1 above)
Comment (isidzukuri) [inline app/controllers/products_controller.rb]: its new if. It wasnt there. You was right

**Pull request merged**